subscriptions-api-openapi.yaml

```diff
@@ -131,120 +138,341 @@ components:
+          type: object
+          additionalProperties:
+            type: string
       required:
+        - id
         - protocol
-        - protocolsettings
-        - sink
     Subscription:
-      properties:
-        id:
-          type: string
-          description: REQUIRED. The unique identifier of the subscription in the scope of the subscription manager.
-          example: 1119920371
-        protocol:
-          type: string
-          enum: ["HTTP", "MQTT3", "MQTT5", "AMQP", "NATS", "KAFKA"]
-          description: REQUIRED. Identifier of a delivery protocol.
-          example: "HTTP"
-        protocolsettings:
-          type: object
-          anyOf:
-            - $ref: "#/components/schemas/HTTPSettings"
-            - $ref: "#/components/schemas/MQTTSettings"
-            - $ref: "#/components/schemas/AMQPSettings"
-            - $ref: "#/components/schemas/NATSSettings"
-            - $ref: "#/components/schemas/ApacheKafkaSettings"
-          description: OPTIONAL. A set of settings specific to the selected delivery protocol provider.
-        sink:
-          type: string
-          format: url
-          description: REQUIRED. The address to which events shall be delivered using the selected protocol.
-          example: "https://endpoint.example.com/webhook"
-        filter:
-          $ref: "#/components/schemas/Filter"
+      type: object
+      title: "Subscription"
+      allOf:
+        - $ref: "#/components/schemas/SubscriptionRequest"
+        - type: object
+          properties:
+            id:
+              type: string
+              description: REQUIRED. The unique identifier of the subscription in the scope of the subscription manager.
+              example: 1119920371
       required:
         - id
-        - protocol
+    Protocol:
+      type: string
+      enum: ["HTTP", "MQTT3", "MQTT5", "AMQP", "NATS", "KAFKA"]
+      description: REQUIRED. Identifier of a delivery protocol.
+      example: "HTTP"
     Filter:
-      properties:
-        dialect:
-          type: string
-          description: Filter dialect
-          example: "simple"
-          default: "simple"
-        conditions:
-          type: array
-          description: Filter conditions
-          items:
-            type: object
+      title: "Filter"
+      description: "A filter from a selection of multiple filter types and dialects"
+      type: object
+      additionalProperties: false
+    AllFilter:
+      allOf:
+        - $ref: "#/components/schemas/Filter"
+        - type: object
+          description: "all filter" 
+          properties:
+            all:
+              description: "This filter evaluates to 'true' if all contained filters are 'true'"
+              type: array
+              minItems: 1
+              items:
+                title: "Filter entry"
+                oneOf:
```

Comment:
I'm not too familiar with openAPI, but does what you define here allow for other filters that are not part of the list below?

Reply:
hmm maybe your "Filter" below covers it?

Reply:
JSON Schema is a bit strange. Yes, the "Filter" here causes this to be polymorphic and to also allow for extension types. 

By all I can tell 🤷‍♂️